### PR TITLE
Added float support to config.lua

### PR DESCRIPTION
**Pull request by EPuncker**
code by @infernumx

### CODE REVIEW

File: src/configmanager.cpp [hunk 1]
```diff
@@ -188,6 +201,8 @@ bool ConfigManager::load()
 	integer[SERVER_SAVE_NOTIFY_DURATION] = getGlobalNumber(L, "serverSaveNotifyDuration", 5);
 	integer[YELL_MINIMUM_LEVEL] = getGlobalNumber(L, "yellMinimumLevel", 2);
 
+	floating[SHARED_EXPERIENCE_MULTIPLIER] = getGlobalFloat(L, "sharedExperienceMultiplier", 1.20);
```
Comment (Shawak) [inline src/configmanager.cpp:204]: 1.2f

Comment (EPuncker) [reply src/configmanager.cpp]: thanks!

**Review by DSpeichert — DISMISSED** [review]

**Comment by peonso** [comment]
While I like it a lot, I'm bothered by consistency, why exp, loot and skill rates aren't floats either? They should and this change might lead people to believe they are.

**Review by danyelvarejao — DISMISSED** [review]

**Comment by nekiro** [comment]
> While I like it a lot, I'm bothered by consistency, why exp, loot and skill rates aren't floats either? They should and this change might lead people to believe they are.

That's great observation. We can update them later in next prs. This one is fine as it is, to me at least.

**Review by nekiro — DISMISSED** [review]

**Review by DSpeichert — APPROVED** [review]

**Comment by DSpeichert** [comment]
I like Float support but I'm not sure if we want to move everything to config.lua.

**Comment by EPuncker** [comment]
> I like Float support but I'm not sure if we want to move everything to config.lua.

that entry was added just as example, should I remove it?

**Comment by DSpeichert** [comment]
> that entry was added just as example, should I remove it?

Hard to say, it's a nice representation of what can be done now, but it could be unnecessary to move things to config.lua and also confusing if other values can't be floats. Maybe remove it and we keep it as a "hidden feature" for now?

**Comment by EPuncker** [comment]
> > that entry was added just as example, should I remove it?
> 
> Hard to say, it's a nice representation of what can be done now, but it could be unnecessary to move things to config.lua and also confusing if other values can't be floats. Maybe remove it and we keep it as a "hidden feature" for now?

done

**Review by DSpeichert — APPROVED** [review]

File: src/configmanager.h [hunk 2]
```diff
@@ -137,12 +137,17 @@ class ConfigManager
 			LAST_INTEGER_CONFIG /* this must be the last one */
 		};
 
+		enum floating_config_t {
+			LAST_FLOATING_CONFIG
```
Comment (nekiro) [inline src/configmanager.h:141]: arrays cant be zero sized, you have to mark this 1 or add at least one config enum

**Review by nekiro — CHANGES_REQUESTED** [review]

**Comment by EPuncker** [comment]
what about we add it when it is useful at all? :D

**Pull request closed without merging**